src/controllers/chat.controller.js

```diff
@@ -62,7 +64,23 @@ exports.createMessage = async (req, res, next) => {
         // add message to messages table
         const insertedMessage = await db.one(queries.createMessage, message);
 
-        // success; return nothing
+        // send request to emoji predictor
+        request.post(
+            emojiApiUrl,
+            { form: { message: req.body.text } },
+            (error, response, body) => {
+                if (!error && response.statusCode == 200) {
+                    const { emoji } = JSON.parse(body);
+                    const recommendedEmoji = [
+                        insertedMessage.id,
+                        emoji
+                    ];
+                    db.none(queries.createRecommendedEmoji, recommendedEmoji);
+                }
+            }
+        );
```

Comment:
I'm not sure if this is the most efficient thing to do every time a message is posted in a chat, but I'm not sure what the alternatives are either. Hm... it's a tough call.

Reply:
yeah if for some reason this gets a ton of usage, it could be optimized by sending the messages in bulk and then receiving/inserting the results in bulk but i think for now it's fine